connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetricsRegistry.java

```diff
@@ -0,0 +1,283 @@
+        sinkRecordOffsetCommitSkipTotal = createTemplate("offset-commit-skip-total", SINK_TASK_GROUP_NAME,
+                                                         "The total number of offset commit completions that were received" +
+                                                         " too late and skipped/ignored.",
+                                                         sinkTaskTags);
+        sinkRecordPutBatchTimeMax = createTemplate("put-batch-max-time-ms", SINK_TASK_GROUP_NAME,
+                                                   "The maximum time taken by this task to put a batch of sinks records.",
+                                                   sinkTaskTags);
+        sinkRecordPutBatchTimeAvg = createTemplate("put-batch-avg-time-ms", SINK_TASK_GROUP_NAME,
+                                                   "The average time taken by this task to put a batch of sinks records.",
+                                                   sinkTaskTags);
+
+        /***** Sink worker task level *****/
+        Set<String> sinkTaskTopicTags = new LinkedHashSet<>(tags);
+        sinkTaskTopicTags.add(CONNECTOR_TAG_NAME);
+        sinkTaskTopicTags.add(TASK_TAG_NAME);
+        sinkTaskTopicTags.add(TOPIC_TAG_NAME);
+        sinkTaskTopicTags.add(PARTITION_TAG_NAME);
+
+        sinkRecordTopicPartitionLag = createTemplate("sink-record-lag", SINK_TASK_GROUP_NAME,
+                                                     "The current lag in terms of number of records behind the consumer the " +
+                                                     "offset commits are for this topic partitions.",
+                                                     sinkTaskTopicTags);
+        sinkRecordTopicPartitionLagMax = createTemplate("sink-record-lag-max", SINK_TASK_GROUP_NAME,
+                                                        "The maximum lag in terms of number of records behind the consumer" +
+                                                        " the offset commits are for this topic partitions.",
+                                                        sinkTaskTopicTags);
+        sinkRecordTopicPartitionLagAvg = createTemplate("sink-record-lag-avg", SINK_TASK_GROUP_NAME,
+                                                        "The average lag in terms of number of records behind the consumer" +
+                                                        " the offset commits are for this topic partitions.",
+                                                        sinkTaskTopicTags);
+    }
+
+    private MetricNameTemplate createTemplate(String name, String group, String doc, Set<String> tags) {
+        MetricNameTemplate template = new MetricNameTemplate(name, group, doc, tags);
+        allTemplates.add(template);
+        return template;
+    }
+
+    public List<MetricNameTemplate> getAllTemplates() {
+        return Collections.unmodifiableList(allTemplates);
+    }
+
+    public String connectorTagName() {
```

Comment:
should we even bother with these accessors since these are public constants anyway?

Reply:
Well, I think it's easier to access the instance method on lines like [this one](https://github.com/apache/kafka/pull/3987/files/9824e329ef599c59ea4e7f60cf46ec907b516d90..d38dbde9f72926c68383729f9c80513879913cde#diff-9d27e74bcdc892150367aed9a4cf499eR655).